File: LC_CPP/14_Longest Common Prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if(strs.empty())
            return "";
        int index = 0;
        int min_len = strs[0].size();
        for(int i = 0; i < strs.size(); i++){
            if(strs[i] == "")
                return "";
            if(strs[i].size() < min_len){
                min_len = strs[i].size();
                index = i;
            }                
        }
        
        for(int i = 0; i < min_len; i++){
            for(int j = 0; j < strs.size(); j++){
                if(strs[j][i] != strs[0][i])
                    return strs[0].substr(0, i);
            }
        }
        return strs[index];        
    }
};

File: Python/14_Longest Common Prefix.py
class Solution:
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        # zip和set的使用
        res = ""
        if len(strs) == 0 :
            return ""
        if len(strs)  == 1:
            return strs[0]
        for each in zip(*strs):
            if(len(set(each)) == 1):
                res += each[0]
            else:
                return res
        return res